WooCommerce/WooCommerceTests/ViewRelated/Products/Media/ProductImageUploaderTests.swift

```diff
@@ -693,6 +731,154 @@ final class ProductImageUploaderTests: XCTestCase {
+                                         isLocalID: false)
+
+        // When - simulate an uploading status
+        mockImageUploader.activeUploadsKeys = [key]
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 1)
+        XCTAssertEqual(activeUploads.first?.siteID, key.siteID)
+        XCTAssertEqual(activeUploads.first?.productOrVariationID, key.productOrVariationID)
+        XCTAssertEqual(activeUploads.first?.isLocalID, key.isLocalID)
+
+        // When - simulate completion of upload
+        mockImageUploader.activeUploadsKeys = []
+
+        // Then
+        XCTAssertEqual(activeUploads.count, 0)
+    }
+
+    func test_background_upload_notice_is_sent_when_there_are_active_uploads_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+
+        let mockImageUploader = MockProductImageUploader()
+
+        let key = ProductImageUploaderKey(siteID: siteID,
+                                         productOrVariationID: productID,
+                                         isLocalID: false)
+
+        let noticePresenter = MockNoticePresenter()
+
+        // When - No active uploads
+        mockImageUploader.activeUploadsKeys = []
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+
+        // Reset flag for next check
+        mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled = false
+
+        // When - With active uploads
+        mockImageUploader.activeUploadsKeys = [key]
+        mockImageUploader.sendBackgroundUploadNoticeIfNeeded(key: key, using: noticePresenter)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.sendBackgroundUploadNoticeIfNeededWasCalled)
+    }
+
+    func test_reset_clears_storage_state_with_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+        let mockImageUploader = MockProductImageUploader()
+
+        // When
+        mockImageUploader.reset()
+
+        // Then
+        XCTAssertTrue(mockImageUploader.resetWasCalled, "Reset should be called")
+    }
+
+    func test_replaceLocalID_updates_storage_when_flag_enabled() {
+        // Given
+        mockFeatureFlagService = MockFeatureFlagService(backgroundProductImageUpload: true)
+        let mockImageUploader = MockProductImageUploader()
+
+        let localProductID: Int64 = 0
+        let remoteProductID = productID.id
+        let localID = ProductOrVariationID.product(id: localProductID)
+
+        // When
+        mockImageUploader.replaceLocalID(siteID: siteID,
+                                        localID: localID,
+                                        remoteID: remoteProductID)
+
+        // Then
+        XCTAssertTrue(mockImageUploader.replaceLocalIDWasCalled, "Replace local ID should be called")
```

Comment:
I don't understand what we are trying to test here. We are firing a method over the `mockImageUploader` and testing that the method is fired. Did you intend to test that storage is updated? 🤔

Reply:
This is a valid observation; as you can see, storage is not included at all here, so I would say that the name is misleading. I renamed it to `test_replaceLocalID_is_called_when_flag_enabled` 12f4c1b0b96e5824bb7175a666a1a98666e548a7